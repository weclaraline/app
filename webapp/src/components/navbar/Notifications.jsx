import React, { useState } from 'react';
import {
  Badge,
  IconButton,
  makeStyles,
  createStyles,
  Popover,
  List,
  ListItem,
  ListItemText,
} from '@material-ui/core';
import { Notifications as NotificationIcon } from '@material-ui/icons';

const useStyles = makeStyles(() =>
  createStyles({
    button: {
      padding: 0,
    },
  })
);

const Notifications = ({ notifications = [] }) => {
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = useState(null);

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <>
      <IconButton
        className={classes.button}
        color="inherit"
        onClick={handleClick}
      >
        <Badge badgeContent={notifications.length} color="error">
          <NotificationIcon />
        </Badge>
      </IconButton>
      {notifications.length > 0 && (
        <Popover
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={handleClose}
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'left',
          }}
          transformOrigin={{
            vertical: 'top',
            horizontal: 'right',
          }}
        >
          <List>
            {notifications.map((notification) => (
              <ListItem key={notification}>
                <ListItemText primary={notification} />
              </ListItem>
            ))}
          </List>
        </Popover>
      )}
    </>
  );
};

export default Notifications;
