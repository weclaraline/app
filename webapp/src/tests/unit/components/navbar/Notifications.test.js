import '../../testing_utils';
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Notifications from "../../../../components/navbar/Notifications";

it("Renders original content", ()=> {
    const tree = shallow( <Notifications />);
    expect(toJson(tree)).toMatchSnapshot();
});