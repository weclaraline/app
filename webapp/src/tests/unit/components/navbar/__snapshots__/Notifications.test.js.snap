// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders original content 1`] = `
<Fragment>
  <WithStyles(ForwardRef(IconButton))
    className="makeStyles-button-1"
    color="inherit"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Badge))
      badgeContent={0}
      color="error"
    >
      <NotificationsIcon />
    </WithStyles(ForwardRef(Badge))>
  </WithStyles(ForwardRef(IconButton))>
</Fragment>
`;
